### PR TITLE
SPECS/cni-plugins: update to v1.3.0 and set version while building

### PR DESCRIPTION
**Pull request by invidian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

To address issues stated in #6339 

###### Change Log  

- cni-plugins has been updated to latest available version v1.3.0 and will now correctly print installed version when binaries are run with `-v`

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Associated issues  

- #6339 

###### Test Methodology

- None

### CODE REVIEW

File: SPECS/cni-plugins/cni-plugins.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 %define _default_cni_plugins_dir /opt/cni/bin
 Summary:        Container Network Interface (CNI) plugins
 Name:           cni-plugins
-Version:        0.9.1
-Release:        14%{?dist}
+Version:        1.3.0
```
Comment (mfrw) [inline SPECS/cni-plugins/cni-plugins.spec:4]: We have missed to update the relevant entry in `cgmanifest.json`

Comment (invidian) [reply SPECS/cni-plugins/cni-plugins.spec:4]: Ops, thanks for pointing out. I had it done locally, but I forgot to push it :)

**Comment by invidian** [comment]
I finally managed to build package locally. Pushed a fix for setting the version. It builds and prints correctly now.

**Review by Camelron — APPROVED** [review]

**Comment by christopherco** [comment]
Buddy Build Queued: [434883](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=434883&view=results)

**Comment by christopherco** [comment]
> Buddy Build Queued: [434883](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=434883&view=results)

Build passed

**Comment by invidian** [comment]
Spec linting is failing because of other packages. Should I fix those as part of this PR?

**Review by romoh — APPROVED** [review]

File: SPECS/cni-plugins/cni-plugins.spec [hunk 2]
```diff
@@ -39,8 +39,14 @@ make -k check |& tee %{_specdir}/%{name}-check-log || %{nocheck}
 %{_default_cni_plugins_dir}/*
 
 %changelog
+<<<<<<< HEAD
```
Comment (christopherco) [inline SPECS/cni-plugins/cni-plugins.spec:42]: The rebase seems to have conflicted here

**Review by christopherco — CHANGES_REQUESTED** [review]
Changes themselves look good but there is a merge conflict to resolve in the spec changelog

**Comment by christopherco** [comment]
> Spec linting is failing because of other packages. Should I fix those as part of this PR?

No, this PR can just focus on the cni-plugins change

File: SPECS/cni-plugins/cni-plugins.spec [hunk 3]
```diff
@@ -39,8 +39,14 @@ make -k check |& tee %{_specdir}/%{name}-check-log || %{nocheck}
...
 * Tue Oct 10 2023 Dan Streetman <ddstreet@ieee.org> - 0.9.1-15
 - Bump release to rebuild with updated version of Go.
+=======
+* Wed Oct 11 2023 Mateusz Gozdek <mgozdek@microsoft.com> - 1.3.0-1
+- Make plugin binaries correctly print version
+- Upgrade to version 1.3.0
+>>>>>>> 5c54ee34c (SPECS/cni-plugins: update to v1.3.0 and set version while building)
```
Comment (christopherco) [inline SPECS/cni-plugins/cni-plugins.spec:49]:
```suggestion
* Wed Oct 11 2023 Mateusz Gozdek <mgozdek@microsoft.com> - 1.3.0-1
- Make plugin binaries correctly print version
- Upgrade to version 1.3.0

* Tue Oct 10 2023 Dan Streetman <ddstreet@ieee.org> - 0.9.1-15
- Bump release to rebuild with updated version of Go.
```

I believe this suggestion should work

**Review by christopherco — APPROVED** [review]

**Pull request merged**